### PR TITLE
MKLDNN INT8 v2 readme.md

### PR DESCRIPTION
**Pull request by bingyanghuang**
This is the first version of int8 v2 readme file, the format is referred to `python/paddle/fluid/contrib/int8_inference/README.md`.

### CODE REVIEW

**Comment by bingyanghuang** [comment]
@wojtuss @sfraczek I've created a readme file draft, if you have any good suggestions, please don't hesitate to leave your comments or rewrite it.

File: paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+# INT8 MKLDNN quantization 
+
+This document describes how to use Paddle inference to convert the FP32 model to INT8 model on ResNet-50 and MobileNet-V1. We provide the instructions on enabling INT8 MKLDNN quantization in Paddle inference and show the ResNet-50 and MobileNet-V1 results in accuracy and performance.
+
+## 0. Prerequisite
+You need to build install at least PaddlePaddle-1.4.The test dataset and model will downloaded automatically in the build process.
```
Comment (luotao1) [inline paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md]: please put the build and install command. You can refer to [1. Install PaddlePaddle]
https://github.com/PaddlePaddle/models/blob/6cf68df0ca9d507a13b76c3531103b1d99cdf3d3/fluid/PaddleCV/image_classification/README_ngraph.md#1-install-paddlepaddle

File: paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+## 1. Enable INT8 MKLDNN quantization
```
Comment (luotao1) [inline paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md]: You can refer to the unit test in xxx.

File: paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+We enable the MKLDNN quantization procedure by analysis config, and pass the warmup data we prepared before. Meanwhile, you can set all the quantization parameters by using quantize config. For example, set quantization op names, set quantization strategies etc.
+
+```cpp
+cfg.EnableMkldnnQuantizer();//Enable MKLDNN quantization
+cfg.mkldnn_quantizer_config()->SetWarmupData(warmup_data); //use analysis config to call the MKLDNN quantization config
+cfg.mkldnn_quantizer_config()->SetWarmupBatchSize(100);
+```
+
+* #### Run the accuracy compare test and profile test
```
Comment (luotao1) [inline paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md]: How about `* #### Run the accuracy compare test and profile test`->`### Run the accuracy compare test and profile test`?
image
the format and indent of `Run the accuracy compare test and profile test` and `To compare the top 1 accuracy before and after MKLDNN quantization
` are exactly same.

File: paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+* The accuracy measurement requires the model with `label`.
+
+## 3. Commands to reproduce the above accuracy and performance benchmark
+
+* #### Small dataset (Single core)
+```bash
+cd /PATH/TO/PADDLE/build
+```
+```bash
+./paddle/fluid/inference/tests/api/test_analyzer_int8_resnet50 --infer_model=/home/bingyang/paddle-latest/build/third_party/inference_demo/int8/resnet50/model --infer_data=/home/bingyang/paddle-latest/build/third_party/inference_demo/int8/data.txt --paddle_num_threads=1
```
Comment (luotao1) [inline paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md]: please remove `/home/bingyang`
```
./paddle/fluid/inference/tests/api/test_analyzer_int8_resnet50 --infer_model=third_party/inference_demo/int8/resnet50/model --infer_data=third_party/inference_demo/int8/data.txt
```
is enough?
Do you need repeat times?

**Review by luotao1 — COMMENTED** [review]
`SetInput` and `CompareQuantizedAndAnalysis` are not official api for EnableMkldnnQuantizer, they are only for tests. The only official api is part of `create quantize config by analysis config`.
Thus, how about only introduce`create quantize config by analysis config` ?

File: paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md (hunk 1 above)
Comment (bingyanghuang) [reply paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md]: Done

File: paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md (hunk 2 above)
Comment (bingyanghuang) [reply paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md]: Done

File: paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md (hunk 3 above)
Comment (bingyanghuang) [reply paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md]: Remove this part

File: paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md (hunk 4 above)
Comment (bingyanghuang) [reply paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md]: Remove /home/bingyang/

Comment (bingyanghuang) [reply paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md]: > please remove `/home/bingyang`
> 
> ```
> ./paddle/fluid/inference/tests/api/test_analyzer_int8_resnet50 --infer_model=third_party/inference_demo/int8/resnet50/model --infer_data=third_party/inference_demo/int8/data.txt
> ```
> is enough?
> Do you need repeat times?

@sfraczek Could you update these commands based on your latest modifications?

File: paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md [hunk 5]
```diff
@@ -2,51 +2,33 @@
...
-
-We enable the MKLDNN quantization procedure by analysis config, and pass the warmup data we prepared before. Meanwhile, you can set all the quantization parameters by using quantize config. For example, set quantization op names, set quantization strategies etc.
-
-```cpp
-cfg.EnableMkldnnQuantizer();//Enable MKLDNN quantization
-cfg.mkldnn_quantizer_config()->SetWarmupData(warmup_data); //use analysis config to call the MKLDNN quantization config
-cfg.mkldnn_quantizer_config()->SetWarmupBatchSize(100);
-```
+## 1. Enable INT8 MKLDNN quantization 
+You can refer to the unit test in `analyzer_int8_image_classification_teser.cc`.
```
Comment (luotao1) [inline paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md]: You can put url after #16399 merged.
teser-》tester

**Review by luotao1 — COMMENTED** [review]
LGTM. Please wait for unit-test and benchmark data.

File: paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+## 0. Install PaddlePaddle 
+Follow PaddlePaddle [installation instruction](https://github.com/PaddlePaddle/models/tree/develop/fluid/PaddleCV/image_classification#installation) to install PaddlePaddle. If you build PaddlePaddle yourself, please use the following cmake arguments. 
+```
+cmake ..  -DWITH_TESTING=ON -WITH_FLUID_ONLY=ON -DWITH_GPU=OFF -DWITH_MKL=ON  -WITH_SWIG_PY=OFF -DWITH_INFERENCE_API_TEST=ON -DON_INFER=ON
+
+```  
+Note: MKLDNN and MKL are required.
+
+## 1. Enable INT8 MKLDNN quantization 
+You can refer to the unit test in `analyzer_int8_image_classification_tester.cc`.
```
Comment (Sand3r-) [inline paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md]: Consider replacing with:
For reference, please examine the code of unit test enclosed in `analyzer_int8_image_classification_tester.cc`.

File: paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+* ### Create Analysis config
+INT8 quantization is one of the optimizations in analysis config. More information about analysis config can be found [here](https://github.com/PaddlePaddle/FluidDoc/blob/develop/doc/fluid/advanced_usage/deploy/inference/native_infer_en.md#upgrade-performance-based-on-contribanalysisconfig-prerelease) 
+
+* ### Create quantize config by analysis config
+
+We enable the MKLDNN quantization procedure by analysis config, and pass the prepared data. Meanwhile, you can set all the quantization parameters by using quantize config. For example, set quantization op names, set quantization strategies etc.
```
Comment (Sand3r-) [inline paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md]: We enable the MKL-DNN quantization procedure by calling an appropriate method from analysis config. Afterwards, all the required quantization parameters (quantization op names, quantization strategies etc.) can be set through quantizer config which is present in the analysis config. It is also necessary to specify a pre-processed warmup dataset and desired batch size.

File: paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+   >**II. Throughput on Intel(R) Xeon(R) Gold 6271 (batch size 1 on single core)**
+
+| Model  | Dataset  | FP32 Throughput  | INT8 Throughput  |  Ratio(INT8/FP32)  |
+| :------------: | :------------: | :------------: | :------------: | :------------: |
+| ResNet-50  | Full ImageNet Val  |  Baidu QA images/s | Baidu QA images/s | Baidu QA |
+| MobileNet-V1 | Full ImageNet Val  | Baidu QA images/s | Baidu QA images/s | Baidu QA  |
+
+Notes:
+* The accuracy measurement requires the model with `label`.
```
Comment (Sand3r-) [inline paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md]: Measurement of accuracy requires a model which accepts two inputs: data and labels.

File: paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+* #### Full dataset (Single core) (WIP)
+```bash
+python preprocess.py --data_dir=/path/to/converted/data.txt
+./paddle/fluid/inference/tests/api/test_analyzer_int8_resnet50 --infer_model=third_party/inference_demo/int8/resnet50/model --infer_data=/path/to/converted/data.txt --paddle_num_threads=1 --test_all_data
+```
+* #### Full dataset (Multi-core) (WIP)
+```bash
+./paddle/fluid/inference/tests/api/test_analyzer_int8_resnet50 --infer_model=third_party/inference_demo/int8/resnet50/model --infer_data=/path/to/converted/data.txt --paddle_num_threads=20 --test_all_data
+```
+   - Notes: This is an example command with 20 cores.
```
Comment (Sand3r-) [inline paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md]: - Notes: This is an example command which utilises 20 cores.

**Review by Sand3r- — COMMENTED** [review]
@bingyanghuang Thank you so much for posting this readme.
Please consider replacing all the MKLDNN occurances to MKL-DNN phrase.

File: paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md (hunk 6 above)
Comment (bingyanghuang) [reply paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md]: Done

File: paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md (hunk 7 above)
Comment (bingyanghuang) [reply paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md]: Done

File: paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md (hunk 9 above)
Comment (bingyanghuang) [reply paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md]: Done

File: paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md (hunk 8 above)
Comment (bingyanghuang) [reply paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md]: Done

**Review by Sand3r- — DISMISSED** [review]

File: paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+## 3. Commands to reproduce the above accuracy and performance benchmark
+
+* #### Small dataset (Single core)
+```bash
+cd /PATH/TO/PADDLE/build
+./paddle/fluid/inference/tests/api/test_analyzer_int8_resnet50 --infer_model=third_party/inference_demo/int8/resnet50/model --infer_data=third_party/inference_demo/int8/data.txt --paddle_num_threads=1 
+```
+* #### Full dataset (Single core) (WIP)
+```bash
+python preprocess.py --data_dir=/path/to/converted/data.txt
```
Comment (luotao1) [inline paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md]: please align line61 with @lidanqing-intel in #16529

File: paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+## 0. Install PaddlePaddle 
+Follow PaddlePaddle [installation instruction](https://github.com/PaddlePaddle/models/tree/develop/fluid/PaddleCV/image_classification#installation) to install PaddlePaddle. If you build PaddlePaddle yourself, please use the following cmake arguments. 
+```
+cmake ..  -DWITH_TESTING=ON -WITH_FLUID_ONLY=ON -DWITH_GPU=OFF -DWITH_MKL=ON  -WITH_SWIG_PY=OFF -DWITH_INFERENCE_API_TEST=ON -DON_INFER=ON
+
+```  
+Note: MKL-DNN and MKL are required.
+
+## 1. Enable INT8 MKL-DNN quantization 
+For reference, please examine the code of unit test enclosed in `analyzer_int8_image_classification_tester.cc`.
```
Comment (luotao1) [inline paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md]: please put url for `analyzer_int8_image_classification_tester.cc`

File: paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md (hunk 10 above)
Comment (lidanqing-vv) [reply paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md]: @bingyanghuang  Please wait a moment. I haven't finish all changes. I changed file name too.

File: paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+./paddle/fluid/inference/tests/api/test_analyzer_int8_resnet50 --infer_model=third_party/inference_demo/int8/resnet50/model --infer_data=third_party/inference_demo/int8/data.txt --paddle_num_threads=1 
+```
+* #### Full dataset (Single core) (WIP)
+```bash
+python preprocess.py --data_dir=/path/to/converted/data.txt
+./paddle/fluid/inference/tests/api/test_analyzer_int8_resnet50 --infer_model=third_party/inference_demo/int8/resnet50/model --infer_data=/path/to/converted/data.txt --paddle_num_threads=1 --test_all_data
+```
+* #### Full dataset (Multi-core) (WIP)
+```bash
+./paddle/fluid/inference/tests/api/test_analyzer_int8_resnet50 --infer_model=third_party/inference_demo/int8/resnet50/model --infer_data=/path/to/converted/data.txt --paddle_num_threads=20 --test_all_data
```
Comment (luotao1) [inline paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md]: Do you need add `--gtest_filter=Analyzer_int8_resnet50.quantization`?

Comment (bingyanghuang) [reply paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md]: Yes, but firstly I will ask Poland team to change the PredictionRun function.

File: paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+
+## 3. Commands to reproduce the above accuracy and performance benchmark
+* #### Small dataset (Single core)
+```bash
+cd /PATH/TO/PADDLE/build
+```
+   * ##### ResNet50 Small dataset benchmark
+```bash
+./paddle/fluid/inference/tests/api/test_analyzer_int8_resnet50 --infer_model=third_party/inference_demo/int8v2/resnet50/model  --infer_data=third_party/inference_demo/int8v2/data.bin --batch_size=1 --paddle_num_threads=1
+```
```
Comment (luotao1) [inline paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md]: 小数据的命令是否可以都挪走？因为上面的表格公布的都是全量数据集的。

File: paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+
+* #### Full dataset (Single core)
+   * ##### Download full ImageNet Validation Dataset
+```bash
+python ../paddle/fluid/inference/tests/api/full_ILSVRC2012_val_preprocess.py
+```
+   * ##### ResNet50 Full dataset benchmark
+```bash
+./paddle/fluid/inference/tests/api/test_analyzer_int8_resnet50 --infer_model=third_party/inference_demo/int8/resnet50/model --infer_data=/path/to/converted/int8_full_val.bin --batch_size=1 --paddle_num_threads=1
```
Comment (luotao1) [inline paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md]: /path/to/converted/ 这个地址是在哪里？
可以说下转换好的数据集默认在/root/.cache/xxxx下面

Comment (bingyanghuang) [reply paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md]: 好的

File: paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md (hunk 13 above)
Comment (bingyanghuang) [reply paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md]: 好的

**Comment by luotao1** [comment]
please pass the travis ci.

**Review by luotao1 — APPROVED** [review]
LGTM. Please cherry-pick to release/1.4. Note that commit info should contain `test=release/1.4`

File: paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md [hunk 15]
```diff
@@ -0,0 +1,70 @@
+# INT8 MKL-DNN quantization 
+
+This document describes how to use Paddle inference Engine to convert the FP32 model to INT8 model on ResNet-50 and MobileNet-V1. We provide the instructions on enabling INT8 MKL-DNN quantization in Paddle inference and show the ResNet-50 and MobileNet-V1 results in accuracy and performance.
+
+## 0. Install PaddlePaddle
```
Comment (mozga-intel) [inline paddle/fluid/inference/tests/api/int8_mkldnn_quantization.md]: 0 -> 1 ?

**Pull request merged**